legacyCode/cacheService.js

```diff
@@ -7,16 +7,37 @@
 
   function getReposFromCache(owner, name) {
     var getKey = owner + '|' + name;
```

Comment:
Now you are generating cache key in 2 places. Please create private method in the module to generate cache key: `_buildKey(owner, repo)` and use in both places.